logger-kafka-consumer/src/main/java/com/trihydro/loggerkafkaconsumer/app/LoggerKafkaConsumer.java

```diff
@@ -75,105 +75,103 @@ public void startKafkaConsumer() throws Exception {
                 ConsumerRecords<String, String> records = stringConsumer.poll(100);
                 recordCount = records.count();
                 if (recordCount > 0) {
-                    utility.logWithDate(String.format("Found %d records to parse", recordCount));
+                    log.info("Found {} records to parse", recordCount);
                 }
                 for (ConsumerRecord<String, String> record : records) {
                     TopicDataWrapper tdw = null;
                     try {
                         tdw = gson.fromJson(record.value(), TopicDataWrapper.class);
                     } catch (Exception e) {
                         // Could be ioException, JsonParseException, JsonMappingException
-                        e.printStackTrace();
+                        log.error("Failed to parse record: {}", record.value(), e);
                     }
                     if (tdw != null && tdw.getData() != null) {
-                        utility.logWithDate(String.format("Found data for topic: %s", tdw.getTopic()));
+                        log.info("Found data for topic: {}", tdw.getTopic());
                         switch (tdw.getTopic()) {
-                        case "topic.OdeTimJson":
-                            utility.logWithDate("Before processing JSON: " + tdw.getData());
-                            odeData = timDataConverter.processTimJson(tdw.getData());
-                            utility.logWithDate(String.format("Parsed TIM: %s", gson.toJson(odeData)));
-                            if (odeData != null) {
-                                if (odeData.getMetadata()
+                            case "topic.OdeTimJson":
+                                log.trace("Before processing JSON: {}", tdw.getData());
+                                odeData = timDataConverter.processTimJson(tdw.getData());
+                                log.trace("After processing JSON: {}", gson.toJson(odeData));
```

Comment:
Note: These log statements were the main cause of excessive logging, overwhelming the console.